chat-widget/src/components/livechatwidget/common/startChat.ts

```diff
@@ -191,15 +190,32 @@ const initStartChat = async (chatSDK: any, chatConfig: ChatConfig | undefined, g
             return;
         }
         // Set app state to failing start chat
-        dispatch({ type: LiveChatWidgetActionType.SET_START_CHAT_FAILING, payload: true});
+        dispatch({ type: LiveChatWidgetActionType.SET_START_CHAT_FAILING, payload: true });
         // Show the loading pane in other cases for failure, this will help for both hideStartChatButton case
         dispatch({ type: LiveChatWidgetActionType.SET_CONVERSATION_STATE, payload: ConversationState.Loading });
+
+        // If sessionInit was successful but LCW startchat failed due to some reason e.g adapter didn't load
+        // we need to directly endChat to avoid leaving ghost chats in OC, not disturbing any other UI state 
+        if (isStartChatSuccessful === true) {
```

Comment:
Shouldn't this be isStartChatSuccessful  === false 
Or !isStartChatSuccessful    ??

Reply:
No, it should be `true` because we're checking if startchat or sessionInit from chatsdk was successful previously, then we endChat.